src/main/java/org/ice4j/socket/SocketReceiveBuffer.java

```diff
@@ -91,102 +81,39 @@ public boolean isEmpty() {
      */
     public boolean offer(DatagramPacket p)
     {
-        boolean added = buffer.offer(p);
-
-        // Keep track of the (total) size in bytes of this receive buffer in
-        // order to be able to enforce SO_RCVBUF.
-        if (added)
+        while (!buffer.offer(p))
```

Comment:
do I understand this loop correctly that it's going to drop packets until there's space available ? Doesn't the API say that it would just return false instead ?

Reply:
Oh I see that in the following commit the method is renamed. Does it make sense to squash the commits ?

Reply:
@paweldomas yes, but usually they were squashed automatically by GitHub when “merge PR” button were hit. Do you want me squash them manually?


> do I understand this loop correctly that it's going to drop packets until there's space available ?


Yes. The oldest packets are dropped. This is what usually done in other places in ice4j and LibJitsi when such overflow is happen.


> Doesn't the API say that it would just return false instead ?
 

It was not handled by caller before and it still unclear what caller can do with that return value - so, I’ve desided to implement the drop behavior on buffer itself.